### ISSUE TITLE
Keycloak integration is broken

### ISSUE THREAD

**Issue by esune** [bcgov/vc-authn-oidc#331]
When completing the demo, redirection to the app fails with this error in Keycloak:
```
vc-authn-keycloak-1       | 2023-09-13 19:59:10,314 ERROR [org.keycloak.broker.oidc.AbstractOAuth2IdentityProvider] (executor-thread-3) Failed to make identity provider oauth callback: org.keycloak.broker.provider.IdentityBrokerException: Could not decode access token response.
vc-authn-keycloak-1       |     at org.keycloak.broker.oidc.OIDCIdentityProvider.getFederatedIdentity(OIDCIdentityProvider.java:370)
vc-authn-keycloak-1       |     at org.keycloak.broker.oidc.AbstractOAuth2IdentityProvider$Endpoint.authResponse(AbstractOAuth2IdentityProvider.java:505)
vc-authn-keycloak-1       |     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
vc-authn-keycloak-1       |     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
vc-authn-keycloak-1       |     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
vc-authn-keycloak-1       |     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:154)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:118)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:560)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:452)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:413)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:415)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:378)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:174)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:142)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:168)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:131)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:33)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:429)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:240)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:154)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:157)
vc-authn-keycloak-1       |     at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:229)
vc-authn-keycloak-1       |     at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:82)
vc-authn-keycloak-1       |     at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:147)
vc-authn-keycloak-1       |     at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:84)
vc-authn-keycloak-1       |     at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:44)
vc-authn-keycloak-1       |     at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
vc-authn-keycloak-1       |     at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
vc-authn-keycloak-1       |     at io.vertx.ext.web.impl.RoutingContextWrapper.next(RoutingContextWrapper.java:200)
vc-authn-keycloak-1       |     at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:58)
vc-authn-keycloak-1       |     at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:36)
vc-authn-keycloak-1       |     at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
vc-authn-keycloak-1       |     at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
vc-authn-keycloak-1       |     at io.vertx.ext.web.impl.RoutingContextWrapper.next(RoutingContextWrapper.java:200)
vc-authn-keycloak-1       |     at org.keycloak.quarkus.runtime.integration.web.QuarkusRequestFilter.lambda$createBlockingHandler$0(QuarkusRequestFilter.java:82)
vc-authn-keycloak-1       |     at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
vc-authn-keycloak-1       |     at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
vc-authn-keycloak-1       |     at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
vc-authn-keycloak-1       |     at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
vc-authn-keycloak-1       |     at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
vc-authn-keycloak-1       |     at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
vc-authn-keycloak-1       |     at java.base/java.lang.Thread.run(Thread.java:833)
vc-authn-keycloak-1       | Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'Internal': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
vc-authn-keycloak-1       |  at [Source: (String)"Internal Server Error"; line: 1, column: 9]
vc-authn-keycloak-1       |     at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2477)
vc-authn-keycloak-1       |     at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:760)
vc-authn-keycloak-1       |     at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:3041)
vc-authn-keycloak-1       |     at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:3019)
vc-authn-keycloak-1       |     at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._matchToken(ReaderBasedJsonParser.java:2793)
vc-authn-keycloak-1       |     at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:2066)
vc-authn-keycloak-1       |     at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:808)
vc-authn-keycloak-1       |     at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4912)
vc-authn-keycloak-1       |     at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4818)
vc-authn-keycloak-1       |     at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3772)
vc-authn-keycloak-1       |     at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3740)
vc-authn-keycloak-1       |     at org.keycloak.util.JsonSerialization.readValue(JsonSerialization.java:75)
vc-authn-keycloak-1       |     at org.keycloak.broker.oidc.OIDCIdentityProvider.getFederatedIdentity(OIDCIdentityProvider.java:368)
vc-authn-keycloak-1       |     ... 44 more
vc-authn-keycloak-1       | 
vc-authn-keycloak-1       | 
vc-authn-keycloak-1       | 2023-09-13 19:59:10,316 WARN  [org.keycloak.events] (executor-thread-3) type=IDENTITY_PROVIDER_LOGIN_ERROR, realmId=vc-authn, clientId=vue-fe, userId=null, ipAddress=172.23.0.1, error=identity_provider_login_failure, code_id=4c0d6cbf-0b62-40c3-adcc-a060ed2ecaaa
```

**Comment by esune** [comment]
The issue actually appears to be with the extraction of attributes from a completed proof, not with Keycloak - see PR for details

**Comment by esune** [comment]
Resolved by #333
